### PR TITLE
Update auth docs

### PR DESCRIPTION
**Pull request by matusf**
Hi, I removed the docs regarding `email/password` auth since it's no longer supported by `python-gitlab`.

### CODE REVIEW

**Comment by max-wittig** [comment]
Thanks for your MR. That's helpful. Just a small comment

File: .gitignore [hunk 1]
```diff
@@ -9,3 +9,6 @@ docs/_build
 .testrepository/
 .tox
 venv/
+.mypy_cache
+.vscode
+
```
Comment (max-wittig) [inline .gitignore]: These are local files, which should be managed by a global gitconfig instead.

Comment (matusf) [reply .gitignore]: Thanks, fixed

**Comment by max-wittig** [comment]
Thanks for the MR!

**Pull request merged**